src/jit/hwintrinsicxarch.cpp

```diff
@@ -292,12 +292,24 @@ GenTree* Compiler::impX86HWIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HA
     }
     else if (!issupported)
     {
-        for (unsigned i = 0; i < sig->numArgs; i++)
+        // When we aren't supported we need to return nullptr when attempting to expand inline
+        // (gtIsRecursiveCall == false) as the gtNewMustThrowException node can mess with the
+        // inliner in unexpected ways.
+
+        if (gtIsRecursiveCall(method))
```

Comment:
Getting back to my comment on your last PR, now that we have a use case -- I'd prefer it if this "recursive call ==> must expand" idiom was confined to the importer; you should pass `mustExpand` down in here to figure out whether bailing out is acceptable behavior.

Reply:
Will update.

Reply:
Fixed. Also see https://github.com/dotnet/coreclr/pull/15772, which has the changes standalone.